File: #942_diStringMatch.py
class Solution:
    def diStringMatch(self, S: str) -> List[int]:
        A = []
        i = 0
        j = len(S)
        for st in S:
            if st == "I":
                A.append(i)
                i += 1
            else:
                A.append(j)
                j -= 1
        A.append(j)
        return A
        

File: Java/#942_diStringMatch.java
class Solution {
    public int[] diStringMatch(String S) {
        int i=0;
        int d=S.length();
        int[] res = new int[S.length()+1];
        for(int j=0;j<S.length();j++){
            if(S.charAt(j)=='I'){
                res[j]=i;
                i++;
            }else{
                res[j]=d;
                d--;
            }
        }
        res[S.length()]=i;
        return res;
    }
}
